src/ORM/ResultSet.php

```diff
@@ -313,10 +313,19 @@ public function first()
      */
     public function serialize()
     {
+        if (!$this->_useBuffering) {
+            $msg = 'You cannot serialize an un-buffered ResultSet. Use Query::bufferResults() to get a buffered ResultSet.';
+            throw new Exception($msg);
+        }
+
         while ($this->valid()) {
             $this->next();
         }
 
+        if ($this->_results instanceof SplFixedArray) {
```

Comment:
Why this broke now and not a year ago is very very strange. I don't think there was anything wrong with using SplFixedArray. There must have been some dependency change in PHP that was across 5.6 and up versions that triggered this. We should report this to the PHP team.